hastexo/hastexo.py

```diff
@@ -232,10 +240,32 @@ def parse_xml(cls, node, runtime, keys, id_generator):
 
         return block
 
-    def author_view(self, context=None):
-        """ Studio View """
-        msg = u"This XBlock only renders content when viewed via the LMS."
-        return Fragment(u'<em>%s</em></p>' % msg)
+    @property
+    def allowed_nested_blocks(self):
+        """
+        Returns a list of allowed nested blocks.
+
+        """
+        additional_blocks = []
+        try:
+            from xmodule.video_module.video_module import VideoDescriptor
+            _spec = NestedXBlockSpec(
+                VideoDescriptor, category="video", label=u"Video"
+            )
+            additional_blocks.append(_spec)
+        except ImportError:
+            pass
+
+        try:
+            from pdf import pdfXBlock
+            _spec = NestedXBlockSpec(pdfXBlock, category="pdf", label=u"PDF")
+            additional_blocks.append(_spec)
+        except ImportError:
+            pass
```

Comment:
Same consideration applies as above for `VideoDescriptor`, though I do realize that the PDF XBlock is an entirely optional add-on component. So, perhaps this should be logged with `logging.info()` or `logging.debug()` instead.